fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/IndexBuilderImpl.java

```diff
@@ -181,6 +182,14 @@ private void indexOcflObject(final String ocflId, final String txId, final OcflO
                 fedoraToOcflObjectIndex.addMapping(txId, fedoraIdentifier, rootFedoraIdentifier, ocflId);
                 LOGGER.debug("Rebuilt fedora-to-ocfl object index entry for {}", fedoraIdentifier);
             });
+
+            headersList.forEach(headers -> {
+                if (!headers.isDeleted()) {
+                    searchIndex.addUpdateIndex(txId, headers);
```

Comment:
Up around L117 you need to clear the search index if there's an exception during the rebuild.